### DISCUSSION TITLE
neo-tree window picker: nvim-window-picker

### DISCUSSION THREAD

**Discussion by krishna-bala** [LazyVim/LazyVim#676] [Q&A]
When using neo-tree, `w` maps to the command `open_with_window_picker`. The default window picker for neo tree appears to be `s1n7ax/nvim-window-picker`, but I don't see that plugin installed with LazyVim (or any other window-picker that could be used). I resolved this by creating the following two files in `lua/plugins`:
```
-- lua/plugins/neo-tree.lua
return {
  "nvim-neo-tree/neo-tree.nvim",
  dependencies = {
    "s1n7ax/nvim-window-picker",
  },
}
```
```
-- lua/plugins/nvim-window-picker.lua
return {
  "s1n7ax/nvim-window-picker",
  config = function()
    require("window-picker").setup()
  end,
}
```

I'm wondering if `nvim-window-picker` should be added to the LazyVim repository so the window picker works out of the box.

**Comment by dpetka2001** [comment]
You could shorten your configuration in a single file like this
```lua
return {
  {
    "nvim-neo-tree/neo-tree.nvim",
    dependencies = {
      "s1n7ax/nvim-window-picker",
      config = function()
        require("window-picker").setup()
      end,
    },
  },
}
```

**Reply by krishna-bala** [reply]
Ah very cool! I didn't realize we can define the config for a dependency like that.

**Reply by folke** [reply]
Instead of `config= ...` you can also just do `opts = {}`

**Reply by krishna-bala** [reply]
`opts = {}` is equivalent to not setting `opts` but setting `config = true`, right?

**Reply by folke** [reply]
opts = {} will call setup. In most cases you dont need `config`

**Comment by nipuna-perera** [comment]
I didn't know this! Does `opts = {}` automatically run the `setup()`
function for the dependency?

**Reply by folke** [reply]
yes, or for any plugin. Best to read the docs of lazy.nvim and LazyVim if you didn't know this

**Comment by pkazmier** [comment]
As a newcomer to LazyVim, I encountered the same issue. Was just coming to see what others use. Thanks for the above.
